File: app/src/test/java/com/mgaetan89/showsrage/adapter/SchedulePagerAdapterTest.java
package com.mgaetan89.showsrage.adapter;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

import com.mgaetan89.showsrage.Constants;
import com.mgaetan89.showsrage.fragment.ScheduleSectionFragment;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class SchedulePagerAdapterTest {
	private SchedulePagerAdapter adapter;

	@NonNull
	private final List<String> ids = Arrays.asList("missed", "today", "soon", "later");

	@NonNull
	private final List<String> labels = Arrays.asList("Missed", "Today", "Soon", "Later");

	@Before
	public void before() {
		this.adapter = new SchedulePagerAdapter(null, this.ids, this.labels);
	}

	@Test
	public void getCount() {
		assertThat(this.adapter.getCount()).isEqualTo(this.ids.size());
	}

	@Test
	public void getItem() {
		for (int i = 0; i < this.ids.size(); i++) {
			Fragment fragment = this.adapter.getItem(i);
			assertThat(fragment).isInstanceOf(ScheduleSectionFragment.class);
			assertThat(fragment.getArguments()).isNotNull();
			assertThat(fragment.getArguments().containsKey(Constants.Bundle.INSTANCE.getSCHEDULE_SECTION()));
		}
	}

	@Test
	public void getPageTitle() {
		for (int i = 0; i < this.ids.size(); i++) {
			assertThat(this.adapter.getPageTitle(i)).isEqualTo(this.labels.get(i));
		}
	}

	@After
	public void after() {
		this.adapter = null;
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/adapter/SchedulePagerAdapterTest.kt
package com.mgaetan89.showsrage.adapter

import android.support.v4.app.FragmentManager
import com.mgaetan89.showsrage.Constants
import com.mgaetan89.showsrage.fragment.ScheduleSectionFragment
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.mock

class SchedulePagerAdapterTest {
	private lateinit var adapter: SchedulePagerAdapter

	private val ids = listOf("missed", "today", "soon", "later")
	private val labels = listOf("Missed", "Today", "Soon", "Later")

	@Before
	fun before() {
		this.adapter = SchedulePagerAdapter(mock(FragmentManager::class.java), this.ids, this.labels)
	}

	@Test
	fun getCount() {
		assertThat(this.adapter.count).isEqualTo(this.ids.size)
	}

	@Test
	fun getItem() {
		for (i in this.ids.indices) {
			val fragment = this.adapter.getItem(i)
			assertThat(fragment).isInstanceOf(ScheduleSectionFragment::class.java)
			assertThat(fragment.arguments).isNotNull()
			assertThat(fragment.arguments!!.containsKey(Constants.Bundle.SCHEDULE_SECTION))
		}
	}

	@Test
	fun getPageTitle() {
		for (i in this.ids.indices) {
			assertThat(this.adapter.getPageTitle(i)).isEqualTo(this.labels[i])
		}
	}
}
